src/Models/RaspberryPiDevice.ts

```diff
@@ -74,118 +77,152 @@ export class RaspberryPiDevice implements Device {
   }
 
   async load(): Promise<boolean> {
-    const deviceFolderPath = this.deviceFolder;
-
-    if (!fs.existsSync(deviceFolderPath)) {
-      throw new Error('Unable to find the device folder inside the project.');
+    if (!fs.existsSync(this.projectFolder)) {
+      throw new Error('Unable to find the project folder.');
     }
 
-    if (!fs.existsSync(path.join(deviceFolderPath, 'node_modules'))) {
-      cp.exec('npm install', {cwd: deviceFolderPath});
-    }
+    this.generateCommonFiles();
+    await this.generateDockerRelatedFiles();
+    await this.generateCppPropertiesFile();
+
     return true;
   }
 
   async create(): Promise<boolean> {
-    const deviceFolderPath = this.deviceFolder;
-
-    if (!fs.existsSync(deviceFolderPath)) {
-      throw new Error('Unable to find the device folder inside the project.');
+    if (!fs.existsSync(this.projectFolder)) {
+      throw new Error('Unable to find the project folder.');
     }
+    
+    this.generateCommonFiles();
+    await this.generateCppPropertiesFile();
+    await this.generateDockerRelatedFiles();
+    await this.generateSketchFile(this.templateFilesInfo);
 
-    try {
-      const iotworkbenchprojectFilePath =
-          path.join(deviceFolderPath, FileNames.iotworkbenchprojectFileName);
-      fs.writeFileSync(iotworkbenchprojectFilePath, ' ');
-    } catch (error) {
-      throw new Error(
-          `Device: create iotworkbenchproject file failed: ${error.message}`);
-    }
+    return true;
+  }
 
-    const vscodeFolderPath =
-        path.join(deviceFolderPath, FileNames.vscodeSettingsFolderName);
-    if (!fs.existsSync(vscodeFolderPath)) {
-      fs.mkdirSync(vscodeFolderPath);
+  // Helper function
+  generateCommonFiles(): void {
+    if (!fs.existsSync(this.vscodeFolderPath)) {
+      fs.mkdirSync(this.vscodeFolderPath);
     }
 
-    const option: vscode.InputBoxOptions = {
-      value: constants.defaultSketchFileName,
-      prompt: `Please input device sketch file name here.`,
-      ignoreFocusOut: true,
-      validateInput: (sketchFileName: string) => {
-        if (!sketchFileName ||
-            /^([a-z_]|[a-z_][-a-z0-9_.]*[a-z0-9_])(\.js)?$/i.test(
-                sketchFileName)) {
-          return '';
-        }
-        return 'Sketch file name can only contain alphanumeric and cannot start with number.';
-      }
-    };
+    if (!fs.existsSync(this.devcontainerFolderPath)) {
+      fs.mkdirSync(this.devcontainerFolderPath);

```

Comment:
should handle the exception from `mkdirSync`.

Reply:
Got.